.circleci/config.yml

```diff
@@ -454,6 +454,45 @@ jobs:
             docker build -t zenithdb/compute-node:latest vendor/postgres && docker push zenithdb/compute-node:latest
             docker tag zenithdb/compute-node:latest zenithdb/compute-node:${DOCKER_TAG} && docker push zenithdb/compute-node:${DOCKER_TAG}
 
+  deploy-staging:
+    docker:
+      - image: cimg/python:3.10
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: Get Zenith binaries
+          command: |
+            rm -rf zenith_install postgres_install.tar.gz zenith_install.tar.gz
+            mkdir zenith_install
+            docker pull --quiet zenithdb/zenith:latest
+            ID=$(docker create zenithdb/zenith:latest)
+            docker cp $ID:/data/postgres_install.tar.gz .
```

Comment:
I think that here we can have a race if two concurrent builds occur. If there are two builds B1 (older) and B2 (newer). B2 may build and push `zenithdb/zenith:latest` faster than B1, then B1 will override it with an older version of image. Because, IIRC, we do not cancel previous build if new one arrives currently.

That way, we should at least use the tag of the current build here, not `latest`, IMO

Reply:
And another thing is that older builds (e.g. after re-run or concurrent run) shouldn't actually override the `latest` if it's newer. I don't know how to check it though

Reply:
> That way, we should at least use the tag of the current build here, not latest, IMO

Hm, and anyway, this doesn't protect us from the situation when we deploy older binaries after newer... I cannot find how to prevent two concurrent workflows to run in parallel in circle. Maybe we can protect from this situation in the console versions update API later

Reply:
May be always use version? That way concurrent build can change docker images order but if during deploy we'll check that we are not downgrading that should be okay.

Reply:
Yes, I meant version in the 'tag of the current build'. And deploy script should check that it isn't deploying something older

Reply:
Race could be only when two PRs merged into main branch at the same time. And yes, CircleCI hasn't ability to prevent concurrent builds. It can cancel redundant builds:

>To save time, you can configure CircleCI to automatically cancel any non-terminated workflows when a newer pipeline is triggered on that same branch.

but I highly not recommend to use this feature. Canceling deploy task in the middle could broke storage nodes (as example - partially deployed safekeepers and not deployed page server and so on).

I'll update CircleCI config to use docker tag in the binaries extraction step.